ocaml/xapi/xapi_xenops.ml

```diff
@@ -2261,6 +2261,14 @@ let resync_resident_on ~__context =
       Db.VM.set_resident_on ~__context ~self:vm ~value:localhost)
       xapi_thinks_are_nowhere;
 
+  List.iter (fun ((id, state), queue_name) ->
+    let xapi_power_state =
```

Comment:
This could be written as:

```
xenapi_of_xenops_power_state (Some state.Vm.power_state)
|> function 
| `Running | `Paused -> add_caches id
| _other -> ()
```

But I'm not sure it is better.

Reply:
I agree that `xapi_power_state` should be eliminated, how about this:
```
match xenapi_of_xenops_power_state (Some state.Vm.power_state) with
| `Running | `Paused -> add_caches id;
| _ -> ()
```